htlcswitch/resolve.go

```diff
@@ -0,0 +1,187 @@
+		//	*caFile = testdata.Path("ca.pem")
+		//}
+		//	creds, err := credentials.NewClientTLSFromFile(*caFile, *serverHostOverride)
+		//	if err != nil {
+		//		log.Fatalf("Failed to create TLS credentials %v", err)
+		//	}
+		//	opts = append(opts, grpc.WithTransportCredentials(creds))
+	} else {
+		opts = append(opts, grpc.WithInsecure())
+	}
+	conn, err := grpc.Dial(cfg.ServerAddr, opts...)
+	if err != nil {
+		log.Errorf("fail to dial: %v", err)
+		return nil, nil, errors.New("fail to open connection to resolver")
+	}
+
+	return conn, pb.NewSwapResolverClient(conn), nil
+}
+
+func queryPreImage(pd *lnwallet.PaymentDescriptor, heightNow uint32) (*pb.ResolveResponse, error) {
+
+	conn, client, err := connectResolver()
+	if err != nil {
+		return nil, err
+	}
+
+	defer conn.Close()
+
+	log.Debugf("Getting pre-image for hash: %v %v for amount %v", pd.RHash, hex.EncodeToString(pd.RHash[:]), int64(pd.Amount))
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	// TODO: add additional attributes needed by the resolver. Like the amount
+	// we got, CTLV, etc
+	resp, err := client.ResolveHash(ctx, &pb.ResolveRequest{
+		Hash:      hex.EncodeToString(pd.RHash[:]),
+		Amount:    int64(pd.Amount),
+		Timeout:   pd.Timeout,
+		Heightnow: heightNow,
+	})
+	if err != nil {
+		log.Errorf("%v.ResolveHash(_) = _, %v: ", client, err)
+		return nil, err
+	}
+	log.Debugf("Got response from Resolver: %v \n", resp)
+	return resp, nil
+}
+
+func update(l *channelLink) {
+	if err := l.updateCommitTx(); err != nil {
+		l.fail(LinkFailureError{code: ErrInternalError},
+			"unable to update commitment: %v", err)
+		return
+	}
+}
+
+func asyncResolve(pd *lnwallet.PaymentDescriptor, l *channelLink, obfuscator ErrorEncrypter, heightNow uint32) {
+
+	go func() {
+
+		// at end we have to update the link with success or failure
+		defer update(l)
+
+		resp, err := queryPreImage(pd, heightNow)
+
+		// if we got an error we fail the HTLC
+		// TODO: provide the specific error back to HTLC source
+		if err != nil {
+			log.Errorf("unable to query invoice registry: "+
+				" %v", err)
+			failure := lnwire.FailUnknownPaymentHash{}
+			l.sendHTLCError(
+				pd.HtlcIndex, failure, obfuscator, pd.SourceRef,
+			)
+			return
+		}
+
+		// we got a pre-image
+		preimage, err := hex.DecodeString(resp.Preimage)
+		if err != nil {
+			log.Errorf("unable to decode Preimage %v : "+
+				" %v", resp.Preimage, err)
+			failure := lnwire.FailUnknownPaymentHash{}
+			l.sendHTLCError(
+				pd.HtlcIndex, failure, obfuscator, pd.SourceRef,
+			)
+			return
+		}
+		var preimageArray [32]byte
+		copy(preimageArray[:], preimage[:32])
+
+		log.Debugf("preimage %v , resp.Preimage %v, preimageArray %v", preimage, resp.Preimage, preimageArray)
+
+		err = l.channel.SettleHTLC(
+			preimageArray, pd.HtlcIndex, pd.SourceRef, nil, nil,
+		)
+		// if there is an error, fail the link
+		if err != nil {
+			//l.fail(LinkFailureError{code: ErrInternalError},
```

Comment:
Can we get rid of these commented out lines?

Reply:
yep, done